src/ain/services/crypto.service.ts

```diff
@@ -3,13 +3,14 @@ import { verify } from 'bitcoinjs-message';
 import { MainNet } from '@defichain/jellyfish-network';
 import { isEthereumAddress } from 'class-validator';
 import { verifyMessage } from 'ethers/lib/utils';
+import { Blockchain } from '../node/node.service';
 
 @Injectable()
 export class CryptoService {
-  public verifySignature(message: string, address: string, signature: string): boolean {
+  public verifySignature(message: string, address: string, signature: string, blockchain: Blockchain): boolean {
```

Comment:
Is there a reason to have blockchain as a parameter? It could be determined from the address.

Reply:
wasn't quite sure, if we want to have it like this. for me both is fine. will adapt that this function retrieves blockchain for itself